### PR TITLE
Removed several event based risk outputs

### PR DESCRIPTION
**Pull request by micheles**
The following outputs are no more stored, they are computed at export time when needed:

1. avg_losses-stats
2. agg_curve-stats
3. loss_curve-stats
4. loss_maps-stats
5. loss_maps-rlzs

Everything is green: https://ci.openquake.org/job/zdevel_oq-engine/2322/

### CODE REVIEW

File: openquake/risklib/scientific.py [hunk 1]
```diff
@@ -1722,6 +1713,49 @@ def _loss_curves(self, assets, mean, mean_averages,
             acc.append(zip(mq_curve, mq_avg))
         return acc  # (N, Q1) triples
 
+    def build_agg_curve_stats(self, loss_curve_dt, dstore):
+        """
+        Build an array `agg_curve-stats`.
```
Comment (ptormene) [inline openquake/risklib/scientific.py]: Please describe the parameters, if needed (or prepend with underscore)

File: openquake/risklib/scientific.py [hunk 2]
```diff
@@ -1525,15 +1465,14 @@ def __init__(self, rlzs, quantiles=()):
         self.quantiles = quantiles
         self.names = ['mean'] + ['quantile-%s' % q for q in quantiles]
 
-    def compute_and_store(self, name, dstore):
+    def compute(self, name, dstore):
         """
         Compute mean and quantiles from the data in the datastore
         under the group `<name>-rlzs` and store them under the group
         `<name>-stats`. Return the number of bytes stored.
```
Comment (ptormene) [inline openquake/risklib/scientific.py]: Please fix the docstring

File: openquake/risklib/scientific.py [hunk 3]
```diff
@@ -950,78 +953,15 @@ def __call__(self, assets, ratios_by_aid, ses_ratio):
                 continue
             counts = numpy.array([(loss_ratios >= ratio).sum(axis=0)
                                   for ratio in self.ratios])
+            poes = build_poes(counts, 1. / ses_ratio)
```
Comment (ptormene) [inline openquake/risklib/scientific.py]: Please add a comment about the fact this is never going to produce a division by zero

File: openquake/risklib/riskinput.py [hunk 4]
```diff
@@ -116,17 +116,20 @@ def __len__(self):
 
     def __toh5__(self):
         attrs = {'time_event': self.time_event or 'None',
-                 'time_events': self.time_events,
-                 'loss_types': hdf5.array_of_vstr(self.loss_types),
-                 'deduc': hdf5.array_of_vstr(self.deduc),
-                 'i_lim': hdf5.array_of_vstr(self.i_lim),
-                 'retro': hdf5.array_of_vstr(self.retro),
+                 'time_events': ' '.join(map(decode, self.time_events)),
```
Comment (ptormene) [inline openquake/risklib/riskinput.py]: Please add a comment about the fact none of the joined items can ever contain spaces

File: openquake/commonlib/writers.py [hunk 5]
```diff
@@ -371,9 +371,12 @@ def write_csv(dest, data, sep=',', fmt='%.6E', header=None, comment=None):
     :param comment:
        optional first line starting with a # character
     """
+    close = True
     if len(data) == 0:
         logging.warn('%s is empty', dest)
-    if not hasattr(dest, 'getvalue'):
+    if hasattr(dest, 'write'):  # file-like object
```
Comment (micheles) [inline openquake/commonlib/writers.py]: Done

File: openquake/risklib/scientific.py (hunk 3 above)
Comment (micheles) [inline openquake/risklib/scientific.py]: I have added an assertion for that

File: openquake/risklib/scientific.py (hunk 2 above)
Comment (micheles) [inline openquake/risklib/scientific.py]: Done

File: openquake/risklib/scientific.py (hunk 1 above)
Comment (micheles) [inline openquake/risklib/scientific.py]: Fixed the docstring

File: openquake/risklib/riskinput.py (hunk 4 above)
Comment (micheles) [inline openquake/risklib/riskinput.py]: Done

**Pull request merged**